scripts/nix-profile.sh.in

```diff
@@ -4,10 +4,34 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
     # This part should be kept in sync with nixpkgs:nixos/modules/programs/shell.nix
 
     NIX_LINK=$HOME/.nix-profile
+    if [ -n "$XDG_DATA_HOME" ]; then
+        NIX_LINK_NEW="$XDG_DATA_HOME/nix/profile"
+    else
+        NIX_LINK_NEW=$HOME/.local/share/nix/profile
+    fi
+    if ! [ -e $NIX_LINK ]; then
+        NIX_LINK="$NIX_LINK_NEW"
+    else
+        if [ -e "$NIX_LINK_NEW" ]; then
+            warning="\033[1;35mwarning:\033[0m"
```

Comment:
I am not sure if we want to check if the terminal supports those escape sequences but I bet it would be safe to assume it does.